ICSharpCode.Decompiler/CSharp/SequencePointBuilder.cs

```diff
@@ -310,6 +316,35 @@ public override void VisitFixedStatement(FixedStatement fixedStatement)
 			VisitAsSequencePoint(fixedStatement.EmbeddedStatement);
 		}
 
+		public override void VisitTryCatchStatement(TryCatchStatement tryCatchStatement)
+		{
+			VisitAsSequencePoint(tryCatchStatement.TryBlock);
+			foreach (var c in tryCatchStatement.CatchClauses) {
+				VisitAsSequencePoint(c);
+			}
+			VisitAsSequencePoint(tryCatchStatement.FinallyBlock);
+		}
+
+		public override void VisitCatchClause(CatchClause catchClause)
+		{
+			StartSequencePoint(catchClause);
+			if (catchClause.Condition.IsNull) {
+				var tryCatchHandler = catchClause.Annotation<TryCatchHandler>();
+				if (!tryCatchHandler.ExceptionSpecifierILRange.IsEmpty) {
+					StartSequencePoint(catchClause.CatchToken);
+					var function = tryCatchHandler.Ancestors.OfType<ILFunction>().FirstOrDefault();
+					AddToSequencePointRaw(function, new[] { tryCatchHandler.ExceptionSpecifierILRange });
+					EndSequencePoint(catchClause.CatchToken.StartLocation, catchClause.RParToken.IsNull ? catchClause.CatchToken.EndLocation : catchClause.RParToken.EndLocation);
+				}
+			} else {
+				StartSequencePoint(catchClause.WhenToken);
+				AddToSequencePoint(catchClause.Condition);
+				EndSequencePoint(catchClause.WhenToken.StartLocation, catchClause.CondRParToken.EndLocation);
+			}
+			catchClause.Body.AcceptVisitor(this);
+			EndSequencePoint(catchClause.StartLocation, catchClause.EndLocation);
```

Comment:
What kind of sequence point is this, going all the way from `catch` to the closing `}` ?
I'd expect one sequence point for `catch (...)` and another for the body.